CrimeMonitor/CrimeMonitor.cs

```diff
@@ -0,0 +1,849 @@
+                configuration.ToFile();
+            }
+            // Make sure the UI is up to date
+            RaiseOnUIThread(RecordUpdatedEvent, criminalrecord);
+        }
+
+        public void readRecord(CrimeMonitorConfiguration configuration = null)
+        {
+            lock (recordLock)
+            {
+                // Obtain current criminal record from configuration
+                configuration = configuration ?? CrimeMonitorConfiguration.FromFile();
+                claims = configuration.claims;
+                fines = configuration.fines;
+                bounties = configuration.bounties;
+                profitShare = configuration.profitShare;
+                homeSystems = configuration.homeSystems;
+                updateDat = configuration.updatedat;
+
+                // Build a new criminal record
+                List<FactionRecord> records = configuration.criminalrecord.OrderBy(c => c.faction).ToList();
+                criminalrecord.Clear();
+                foreach (FactionRecord record in records)
+                {
+                    criminalrecord.Add(record);
+                }
+            }
+        }
+
+        private FactionRecord AddRecord(string faction)
+        {
+            if (faction == null) { return null; }
+
+            FactionRecord record = new FactionRecord(faction);
+            GetFactionData(record);
+
+            lock (recordLock)
+            {
+                criminalrecord.Add(record);
+            }
+            return record;
+        }
+
+        private void RemoveRecord(FactionRecord record)
+        {
+            // Check if claims or crimes are pending
+            if ((record.factionReports?.Any() ?? false)) { return; }
+            {
+                _RemoveRecord(record);
+            }
+        }
+
+        public void _RemoveRecord(FactionRecord record)
+        {
+            string faction = record.faction.ToLowerInvariant();
+            lock (recordLock)
+            {
+                for (int i = 0; i < criminalrecord.Count; i++)
+                {
+                    if (criminalrecord[i].faction.ToLowerInvariant() == faction)
+                    {
+                        criminalrecord.RemoveAt(i);
+                        break;
+                    }
+                }
+            }
+        }
+
+        public FactionRecord GetRecordWithFaction(string faction)
+        {
+            if (faction == null)
+            {
+                return null;
+            }
+            return criminalrecord.FirstOrDefault(c => c.faction.ToLowerInvariant() == faction.ToLowerInvariant());
+        }
+
+        public StarSystem GetFactionSystem(string faction, int sphereLy = 20)
+        {
+            if (faction == null) { return null; }
+
+            string currentSystem = EDDI.Instance?.CurrentStarSystem?.name;
+            List<Dictionary<string, object>> sphereSystems = StarMapService.GetStarMapSystemsSphere(currentSystem, 0, sphereLy);
```

Comment:
Use the cube method? Nicer to EDSM servers and a wider net for us too.

Reply:
I'm not 'using' either. I wrote both a cube and a sphere method to test their viability and settled on cube method. Later, your new `GetFactionByName` method negated the need for either.

I just haven't stripped it out yet... Didn't see the urgent need.

Reply:
Removed.